### PR TITLE
Integrate Server with DB

### PR DESCRIPTION
**Pull request by jabrake**
- Created functions and helpers to select/insert/update data in DB and return when appropriate to client
- Made some optimizations to queries from before
- Using environment variables now for DB connection

It's a big one, sorry! @farhanali3193 @tchitrakorn

### CODE REVIEW

File: db/models.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  if (sort === 'newest') {
+    sortValue = 'date DESC';
+  } else if (sort === 'helpful') {
+    sortValue = 'helpfulness DESC';
+  } else if (sort === 'relevant') {
+    sortValue = 'date DESC, helpfulness DESC';
+  } else { console.log('invalid sort value'); }
+
+  const query = {
+    text: `SELECT * FROM reviews WHERE product_id = $1 AND reported = false ORDER BY ${sortValue} OFFSET $2 LIMIT $3`,
```
Comment (farhanali3193) [inline db/models.js]: @jabrake : Shouldn't sortValue be also parameterized like the other dynamic variables such as productId and sort?

File: db/models.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+const updateCharacteristicsReviewsTable = (reviewId, characteristicsObject) => {
+  // eslint-disable-next-line max-len
+  const updateCharacteristicsReviewsQuery = helpers.createUpdateCharacteristicsReviewsQuery(reviewId, characteristicsObject);
+  return client.query(updateCharacteristicsReviewsQuery)
+    .then((res) => console.log('Inserted rows:', res.rowCount))
+    .catch((error) => error.stack);
+};
+
+async function postReview(reviewDetails) {
```
Comment (farhanali3193) [inline db/models.js:74]: @jabrake : After looking at your code, I just realized I also used async await in my db module, and I also forgot to use the try/catch block.

File: db/models.js (hunk 1 above)
Comment (jabrake) [reply db/models.js]: Good catch - updated!

File: db/models.js (hunk 2 above)
Comment (jabrake) [reply db/models.js:74]: Thanks for this one too, updated!

**Review by farhanali3193 — APPROVED** [review]

**Pull request merged**